### ISSUE TITLE
window_focus_cycle request

### ISSUE THREAD

**Issue by tschmidt64-SC** [koekeishiya/chunkwm#456]
I have the following setup:
```
    Monitor 1           Monitor 2
=================   =================
|| APP1 | APP2 ||   || APP3 | APP4 ||
=================   =================
```
(and have window_focus_cycle set to `all`)

Starting at APP1, and repeatedly running `tiling::window --focus east`, I would expect the order of apps to be:
`1 -> 2 -> 3 -> 4 -> 1 -> 2 -> 3 -> 4 -> etc etc etc`
But instead, the behavior is:
`1 -> 2 -> 3 -> 4 -> 2 -> 3 -> 4 -> 2 -> 3 -> 4 -> 2 `

I can understand why this behavior is the way that it is, but it seems somewhat unnatural when you think about how wrapping monitors usually works. Is there any way of changing this?

**Comment by koekeishiya** [comment]
The `window_focus_cycle` all internally calls `focus_monitor` when it wraps to another monitor, which will focus the window on said monitor that last had focus (default macOS behaviour).

**Comment by tschmidt64-SC** [comment]
Yes and that makes sense for certain use cases (if you use multiple desktops for example), but not as much for people who use multi-monitor setups as if they were just one large monitor, where you would expect wrapping to just go from 4 to 1. 

i3, for example, wraps spatially from right to left instead of temporally from right to most recently focused

**Comment by dominiklohmann** [comment]
I'm not sure when this changed, but this default macOS behaviour now breaks mouse follows focus. 

display 1: app A
display 2: app B
display 3: app A

When cycling window focus (to the next window) the mouse ends up on display 3, but the focus jumps to display 1 when coming from display 2. Honestly a major annoyance especially with triple monitors since this keeps getting worse.

**Comment by koekeishiya** [comment]
I get why this is useful, but the reason it works as it does currently, is because of the fact that I have not found a way around the focusing issue presented in #153 (aka default macOS behaviour).

TL;DR

If you have a window of application A open on monitor 1, and another window of application A on monitor 3. The currently focused window of application A is the one on monitor 3.

Now you have focus application B on monitor 2 to do some stuff. You are finished and want to do some stuff in the window of application A on monitor 1, so you try to focus this window. THIS WILL FAIL because macOS focues application A on monitor 3, because that's the window that last had focus for application A.

This still happens with the behaviour we have today, but it is not so disturbing when it happens, because mentally you are not expecting the "window to the west" to gain focus.

I have tried a combination of steps to bypass this behaviour, but I don't think it is possible. It is such a core part of how applications work on macOS.

A sidenote that is not really relevant here, but the reason why we cannot also simply remove titlebars (window decorations) from all applications on macOS as well, is that the application/window itself is the one responsible for setting up the titlebar, and not macOS' WindowServer.

**Comment by EvanCarroll** [comment]
So basically the tldr is that if I have a multi monitor setup and I have a terminal open on one monitor then regardless of the monitor in focus when I spawn a new terminal with Cmd + Return it can open up any monitor with the terminal (the first one), and not the current one I have focused?
